packages/flutter/lib/src/gestures/mouse_tracking.dart

```diff
@@ -0,0 +1,308 @@
+    final String none = (onEnter == null && onExit == null && onMove == null) ? '<none>' : '';
+    return '[$runtimeType${hashCode.toRadixString(16)} $none'
+        '${onEnter == null ? '' : 'onEnter '}'
+        '${onMove == null ? '' : 'onMove '}'
+        '${onExit == null ? '' : 'onExit'}]';
+  }
+}
+
+// Used for accounting for which annotation is active inside of the MouseTracker.
+class _TrackedAnnotation {
+  _TrackedAnnotation(this.annotation);
+
+  final MouseDetectorAnnotation annotation;
+
+  /// True if the mouse pointer is currently inside of the annotated layer. Used
+  /// to detect layers that used to have the mouse pointer inside them, but now
+  /// no longer do (to facilitate exit notification).
+  bool active = false;
+}
+
+/// Describes a function that finds an annotation given an offset in logical
+/// coordinates.
+typedef MouseDetectorAnnotationFinder = MouseDetectorAnnotation Function(Offset offset);
+
+/// Keeps state about which objects are interested in tracking mouse positions
+/// and notifies them when a mouse pointer enters, moves, or leaves an annotated
+/// region that they are interested in.
+///
+/// Owned by the [GestureBinding] class.
+class MouseTracker {
+  /// Creates a mouse tracker to keep track of mouse locations.
+  ///
+  /// All of the parameters must not be null.
+  MouseTracker(PointerRouter router, this.annotationFinder)
+      : assert(router != null),
+        assert(annotationFinder != null) {
+    router.addGlobalRoute(_handleEvent);
+  }
+
+  /// Used to find annotations at a given logical coordinate.
+  final MouseDetectorAnnotationFinder annotationFinder;
+
+  // The set of annotations that are currently being tracked. They may or may
+  // not be active, depending on the value of _TrackedAnnotation.active.
+  final Set<_TrackedAnnotation> _trackedAnnotations = Set<_TrackedAnnotation>();
+
+  /// Allows the [MouseDetector] to attach an annotation as an active
+  /// annotation.
+  void attachAnnotation(MouseDetectorAnnotation annotation) {
+    _trackedAnnotations.add(_TrackedAnnotation(annotation));
+    // Schedule a check so that we test this new annotation to see if the mouse
+    // is currently inside its region.
+    _scheduleMousePositionCheck();
+  }
+
+  /// Allows the [MouseDetector] to detach an annotation as an active annotation.
+  void detachAnnotation(MouseDetectorAnnotation annotation) {
+    final _TrackedAnnotation tracked = _findAnnotation(annotation);
+    assert(tracked != null, "Tried to detach an annotation that wasn't attached: $annotation");
+    if (tracked.active) {
+      annotation.onExit(MouseExitDetails(
+          globalPosition: lastMouseEvent?.position,
+          sourceTimeStamp: lastMouseEvent?.timeStamp));
+    }
+    _trackedAnnotations.remove(tracked);
+  }
+
+  void _scheduleMousePositionCheck() {
+    SchedulerBinding.instance.addPostFrameCallback((Duration _) => collectMousePositions());
+    SchedulerBinding.instance.scheduleFrame();
```

Comment:
it's a pity if there was no frame scheduled that we have to wait an average of 24ms to get a result...